sql/core/src/test/scala/org/apache/spark/sql/connector/WriteDistributionAndOrderingSuite.scala

```diff
@@ -53,15 +56,36 @@ class WriteDistributionAndOrderingSuite extends DistributionAndOrderingSuiteBase
     .add("data", StringType)
 
   test("ordered distribution and sort with same exprs: append") {
-    checkOrderedDistributionAndSortWithSameExprs("append")
+    Seq(true, false).foreach { distributionStrictlyRequired =>
+      Seq(true, false).foreach { dataSkewed =>
+        Seq(true, false).foreach { coalesce =>
```

Comment:
this pattern appears many times, can we deduplicate the code?
```
def checkOrderedDistributionAndSortWithSameExprs(...) {
  Seq(true, false).foreach...
  ...
      checkOrderedDistributionAndSortWithSameExprs(..., distributionStrictlyRequired, dataSkewed, coalesce)
}
```

Reply:
Fixed. Thanks!